### DISCUSSION TITLE
Significant ghosting using book cover as the sleep screen (Kobo Clara 2E)?

### DISCUSSION THREAD

**Discussion by Chairzard** [koreader/koreader#12950] [Q&A]
I've noticed that using a book cover on my Sleep Screen on KOReader on my Kobo Clara 2E results in significantly more ghosting than doing so in the stock reader, to the point that I can basically read the page underneath the cover sometimes depending on the colors used in the cover (see the images below to compare, 1 from the stock reader and 1 from KOReader). It looks like the stock reader does an extra set of flashes when the image is loaded upon entering sleep to clear artifacts from the screen, but KOReader does not.

Is there any way to force KOReader to perform an extra set of flashes myself after the cover is on the screen, is this something I should consider submitting as a feature request (example: creating an option in the sleep or screen section to do an extra flash of the screen upon entering sleep), or is this something that would not be possible due to hardware/driver-based limitations?

**KOReader sleep screen** (below)

![KOReader Sleep](https://github.com/user-attachments/assets/12c50006-700a-4a18-88aa-d9123a5e360c)

**Kobo stock reader sleep screen** (below)

![Kobo Stock Sleep](https://github.com/user-attachments/assets/a3157df4-e37b-4c0c-88a9-197041bc330e)

Link to the book in question if anybody wants to play around with it: https://www.gutenberg.org/ebooks/537

**Comment by poire-z** [comment]
On my Kobo GloHD, it does display/flash a full white background (where we don't see the book text at all) before showing the book cover - at which point only the backlight turns off.
Don't you see it ? If not, it may be some other option at play like border fill or something else.
Or if you do, do you see the book text remnants on it ?

**Reply by Frenzie** [reply]
Same on my H2O and Libra 2 (full flash).

**Reply by Chairzard** [reply]
Border fill seems to be playing some role. I was using black border fill in my screenshots:
- White color fill is about as bad as black color fill.
- With border fill set to "no fill", the ghosting patterns are different. Sometimes it looks better, other times I see ghosting of white text rather than black (it depends on the book cover).

I've attached a video (slowed down) showing it going to sleep in KOReader with black border fill (sorry for the strobing LED lights above!).

https://github.com/user-attachments/assets/c12c3426-c341-40ba-ae96-4deac115735e

**Reply by NiLuJe** [reply]
As was mentioned earlier, we should be doing an extra flash to white beforehand.

The MTK driver might be fucking that one up here, l'll have to double-check.

(It might also simply be a timing issue, sometimes a flash to white beforehand will actually *hurt* ghosting prevention, depending on the delay between all of the refreshes involved...).

**Reply by ahmadadam96** [reply]
Did anyone find a solution to this? I'm still facing the same issue. The sleep covers look so much better in the default OS

**Reply by Luuxuu** [reply]
Still an issue sadly... Any solutions yet?

**Reply by greentea43** [reply]
I am fighting this problem on multiple Kindles throughout the years (I use font weight rather pumped up). There is a flash to white, but then it flashes back to book page, just before the screensaver. If someone can come up with a patch to flash the screen to full black and stay that way until screensaver, would it help?

**Reply by Luuxuu** [reply]
> I am fighting this problem on multiple Kindles throughout the years (I use font weight rather pumped up). There is a flash to white, but then it flashes back to book page, just before the screensaver. If someone can come up with a patch to flash the screen to full black and stay that way until screensaver, would it help?

I think so yeah, mimicking the native OS' refresh would help (full white, then full black, iirc)

**Reply by ghost** [reply]
https://github.com/koreader/koreader/pull/14589/ might fix the ghosting after a resume

**Reply by greentea43** [reply]
> #14589 might fix the ghosting after a resume

How  is this related? We're talking ghosting on suspend here...

**Reply by ghost** [reply]
Sorry, this is a differnt fix.

**Reply by rociojgo** [reply]
Any news on this issue? I've been experiencing the same in my Kindle 11gen too

**Reply by sharonrcg** [reply]
also experiencing this on my 12th gen kindle ppw

**Comment by yparitcher** [comment]
I dont usualy see discussions, so sorry for the late answer,
Do you have the kindle Page Turn Animations active?

**Reply by greentea43** [reply]
> I dont usualy see discussions, so sorry for the late answer, Do you have the kindle Page Turn Animations active?

No Page Turn Animations in my case, PW5. Also, as someone mentioned earlier, Amazon OS sleep screen does not ghost at all...

**Reply by rociojgo** [reply]
Second what @greentea43 mentions, no page turn animations active and the amazon sleep screen doesn't ghost, it just happens in Koreader for some reason (Kindle 11gen).
